src/Adapter/Product/QueryHandler/AbstractSupplierOptionsHandler.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\QueryHandler;
+
+use PrestaShop\PrestaShop\Adapter\Product\Repository\ProductSupplierRepository;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\ValueObject\CombinationId;
+use PrestaShop\PrestaShop\Core\Domain\Product\Supplier\QueryResult\ProductSupplierForEditing;
+use PrestaShop\PrestaShop\Core\Domain\Product\Supplier\QueryResult\ProductSupplierInfo;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+
+/**
+ * Holds reusable methods for querying product supplier
+ */
+abstract class AbstractSupplierOptionsHandler
+{
+    /**
+     * @var ProductSupplierRepository
+     */
+    protected $productSupplierRepository;
+
+    public function __construct(
+        ProductSupplierRepository $productSupplierRepository
+    ) {
+        $this->productSupplierRepository = $productSupplierRepository;
+    }
+
+    /**
+     * @param ProductId $productId
+     * @param CombinationId|null $combinationId
+     *
+     * @return ProductSupplierInfo[]
```

Comment:
```suggestion
     * @return array<int, ProductSupplierInfo>
```

Reply:
although I think its quite clear as it is, but I'll update it and I suppose this means its a new convention? - we use psalm notations everywhere (and not `Object[]`) :thinking:

Reply:
Yeah I'm not sure this notation brings out much values 🤷‍♂️
For more advanced array format it can be very valuable but in this case it's just a simple array of objects What's the interest of this despite making PHPStan happy maybe? If it's just for PHPStan I don't think we should let him dictate what we need to do if it doesn't bring any value for us
Don't let robots control you!! 😜

Reply:
php stan is ok with previous notation.

Reply:
so, @Progi1984 , @jolelievre , @matks could we agree on which one to use when we have simple array?

Reply:
btw I think it would make sense to use `ProductSupplierInfo[]` notations where its simple array (index starts at 0) and maybe array<int, ProductSupplierInfo> in case array keys are ids or some other custom numbers (the idea would be to point out that its not ordinary php array starting from 0).

Reply:
Please always use Psalm notations, it will be required in PHPStan level 6.

Reply:
oh, well okay then